packages/battery/android/src/main/java/io/flutter/plugins/battery/BatteryPlugin.java

```diff
@@ -22,25 +24,42 @@
 import io.flutter.plugin.common.PluginRegistry;
 
 /** BatteryPlugin */
-public class BatteryPlugin implements MethodCallHandler, StreamHandler {
+public class BatteryPlugin implements MethodCallHandler, StreamHandler, FlutterPlugin {
+
+  private Context applicationContext;
+  private BroadcastReceiver chargingStateChangeReceiver;
+  private MethodChannel methodChannel;
+  private EventChannel eventChannel;
 
   /** Plugin registration. */
   public static void registerWith(PluginRegistry.Registrar registrar) {
-    final MethodChannel methodChannel =
-        new MethodChannel(registrar.messenger(), "plugins.flutter.io/battery");
-    final EventChannel eventChannel =
-        new EventChannel(registrar.messenger(), "plugins.flutter.io/charging");
-    final BatteryPlugin instance = new BatteryPlugin(registrar);
-    eventChannel.setStreamHandler(instance);
-    methodChannel.setMethodCallHandler(instance);
+    final BatteryPlugin instance = new BatteryPlugin();
+    instance.onAttachedToEngine(registrar.context(), registrar.messenger());
   }
 
-  BatteryPlugin(PluginRegistry.Registrar registrar) {
-    this.registrar = registrar;
+  @Override
+  public void onAttachedToEngine(FlutterPluginBinding binding) {
+    applicationContext = binding.getApplicationContext();
```

Comment:
I think this is redundant with `this.applicationContext = applicationContext;` in the other `onAttachedToEngine`